cmd/apply.go

```diff
@@ -36,32 +30,39 @@ and see the samples dir`
 			validateToken(context.Background())
 		},
 		PreRunE: func(cmd *cobra.Command, args []string) error {
-			return validateFlags(cmd, args)
+			return validateFlags()
 		},
 		Run: func(cmd *cobra.Command, args []string) {
-			runApplyCmd(context.Background())
+			configHandler, err := newConfigurationHandler(fileToApply)
+			if err != nil {
+				fmt.Println(err)
+			}
+			err = configHandler(context.Background(), applyResource)
+			if err != nil {
+				fmt.Printf("Failed to apply resource - %s\n", err.Error())
+			}
 		},
 	}
 )
 
-func runApplyCmd(ctx context.Context) {
-	fileExtension := filepath.Ext(fileToApply)[1:]
+func newConfigurationHandler(filepath string) (configHandler, error) {
+	fileExtension := fp.Ext(filepath)[1:]
 
 	switch fileExtension {
 	case "json":
-		err := handleJsonFile(ctx)
-		if err != nil {
-			fmt.Printf("Failed to apply resource - %s\n", err.Error())
-		}
+		return handleJsonFile, nil
 	case "yaml", "yml":
-		err := handleYamlFile(ctx)
-		if err != nil {
-			fmt.Printf("Failed to apply resource - %s\n", err.Error())
-		}
+		return handleYamlFile, nil
+	default:
+		return nil, errors.New("wrong file extension: JSON and YAML formats are accepted only")
 	}
 }
 
-func handleYamlFile(ctx context.Context) error {
+type configHandler func(context.Context, commandHandler) error
```

Comment:
refactor the generic part that all three cmds (apply, create and edit) are using into different class